core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -14,180 +14,233 @@
-//    private final Timestamp start;
-//
-//    private MutableDistributionViewData(
-//        DistributionView distributionView, Timestamp start) {
-//      this.distributionView = distributionView;
-//      this.start = start;
-//    }
-//
-//    private final List<Tag> generateTags(List<TagValue> tagValues) {
-//      final List<Tag> tags = new ArrayList<Tag>(tagValues.size());
-//      int i = 0;
-//      for (TagKey tagKey : this.distributionView.getColumns()) {
-//        tags.add(Tag.create(tagKey, tagValues.get(i)));
-//        ++i;
-//      }
-//      return tags;
-//    }
-//
-//    // TODO(songya): remove DistributionAggregate.Range, then remove this method
-//    private static final DistributionAggregate.Range convertRange(
-//        MutableDistribution.Range range) {
-//      return DistributionAggregate.Range.create(range.getMin(), range.getMax());
-//    }
-//  }
-//
-//  /**
-//   * A {@link MutableViewData} for recording stats on interval-based aggregations.
-//   */
-//  static final class MutableIntervalViewData extends MutableViewData {
-//
-//    /**
-//     * Constructs a new {@link MutableIntervalViewData}.
-//     */
-//    static MutableIntervalViewData create(IntervalView view) {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//
-//    @Override
-//    View getView() {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//
-//    @Override
-//    void record(StatsContextImpl tags, double value) {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//
-//    @Override
-//    void record(StatsContextImpl tags, long value) {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//
-//    @Override
-//    final ViewData toViewData(Clock clock) {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//  }
+  @VisibleForTesting
+  static List<TagValue> getTagValues(Map<TagKey, TagValue> tags, List<TagKey> columns) {
+    List<TagValue> tagValues = new ArrayList<TagValue>(columns.size());
+    // Record all the measures in a "Greedy" way.
+    // Every view aggregates every measure. This is similar to doing a GROUPBY view’s keys.
+    for (int i = 0; i < columns.size(); ++i) {
+      TagKey tagKey = columns.get(i);
+      if (!tags.containsKey(tagKey)) {
+        // replace not found key values by “unknown/not set”.
+        tagValues.add(UNKNOWN_TAG_VALUE);
+      } else {
+        tagValues.add(tags.get(tagKey));
+      }
+    }
+    return tagValues;
+  }
+
+  /**
+   * Create an empty {@link MutableAggregation} based on the given {@link Aggregation}.
+   *
+   * @param aggregation {@code Aggregation}.
+   * @return an empty {@code MutableAggregation}.
+   */
+  @VisibleForTesting
+  static MutableAggregation createMutableAggregation(Aggregation aggregation) {
+    return aggregation.match(
+        new Function<Sum, MutableAggregation>() {
```

Comment:
Should these functions be static so we don't need to create a new instance each time we create a mutable aggregation?